### PR TITLE
Portal 1514/status tracker component

### PR DESCRIPTION
**Pull request by ZacharyRizer**
Related to this ticket: https://cdc-dexops.atlassian.net/browse/PORTAL-1514

* created the `ProgressTracker` component
* design aligned with this figma: https://www.figma.com/file/TrW6WMf6GYgscPkVQaiMZ0/CDC-React-Components?type=design&node-id=1887-4672&mode=design&t=mT0ftvVfPshafIa1-0
* created stories for the component to show the three different variants (inProgress-Determinate, inProgress-Indeterminate, and Complete)
* added tests for the component

### CODE REVIEW

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx
Comment (sublettk) [inline packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:1]: Observation from Storybook:
- `currentAmount` should maintain a whole number as it increases and decreases
- The value of `currentAmount` should be between 0 and 100.
- `totalAmount` should be between 0 and 100.

**Review by sublettk — CHANGES_REQUESTED** [review]
Looks good so far. There are some changes that need to be revised regarding the boundaries of the props.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+export interface ProgressTrackerProps {
+  complete?: boolean;
+  currentAmount?: number;
+  indeterminate?: boolean;
+  label?: string;
+  totalAmount?: number;
+}
+
+export const ProgressTracker = ({
+  complete = false,
```
[the remark this answers is not in the document]
Comment (ZacharyRizer) [reply packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:12]: Yep, good call.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  currentAmount = 0,
+  indeterminate = false,
```
[the remark this answers is not in the document]
Comment (ZacharyRizer) [reply packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:14]: Yep, agreed.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx
Comment (ZacharyRizer) [reply packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:1]: Adjusted handling of current vs total amount. If current amount exceeds total, current is set to the total and `100%` is returned. Percentage calculations always `floor` to an integer.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return classList.join(" ");
+  };
+
+  const getBarPercentage = (): number => {
+    if (indeterminate || complete) return 100;
+    return totalAmount > 0 ? (currentAmount / totalAmount) * 100 : 0;
+  };
+
+  const getBarPercentageInfo = (): string => {
+    if (complete) return "Complete";
```
[the remark this answers is not in the document]
Comment (ZacharyRizer) [reply packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx]: I decided to leave the `currentAmount` alone in the case of the `isComplete` boolean. If we need to adjust down the line we can, but I don't want to manipulate internal state unless needed. I think having transparency and data consistency is more important here.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.test.tsx
[the remark this answers is not in the document]
Comment (ZacharyRizer) [reply packages/cdc-react/src/components/ProgressTracker/ProgressTracker.test.tsx:1]: Tests added!

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.scss [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  font-family: global.$font-family-sans;
+  font-size: global.$font-size-small;
+  font-weight: global.$font-weight-light;
+  padding-bottom: 0.5rem;
+}
+
+.progress-bar-background {
+  width: 100%;
+  background-color: global.$gray-light;
+  border-radius: 8px;
```
Comment (ian-sawyer) [inline packages/cdc-react/src/components/ProgressTracker/ProgressTracker.scss:31]: I think this should be 1rem to align more with the design.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.scss [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  overflow: hidden;
+}
+
+.progress-bar-fill {
+  height: 20px;
+  background-color: global.$main-theme;
+  text-align: right;
+  border-radius: 8px;
```
Comment (ian-sawyer) [inline packages/cdc-react/src/components/ProgressTracker/ProgressTracker.scss:39]: I think this should be 1rem to align more with the design.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const getBarPercentage = (): number => {
+    if (isIndeterminate || isComplete) return 100;
+
+    currentAmount = Math.max(0, Math.min(currentAmount, totalAmount));
+    totalAmount = Math.max(0, totalAmount);
+
+    const percentage =
+      totalAmount > 0 ? (currentAmount / totalAmount) * 100 : 0;
+
+    return Math.floor(percentage);
```
Comment (ian-sawyer) [inline packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:35]: I think making this `Math.floor(percentage).toFixed(2);` would be best for display purposes.

File: packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  const getBarPercentageInfo = (): string => {
+    if (isComplete) return "Complete";
+    if (isIndeterminate) return "In progress...";
+    return `${getBarPercentage()}% complete`;
```
Comment (ian-sawyer) [inline packages/cdc-react/src/components/ProgressTracker/ProgressTracker.tsx:41]: A wrapper method to slice the string max to show two decimals would be useful here I think.

**Review by ian-sawyer — APPROVED** [review]
LGTM!

**Review by sublettk — APPROVED** [review]
LGTM!

**Pull request merged**